File: Data Structure/LinkedList/Insert a Node at a specific Position/sol.java
/*
  Insert Node at a given position in a linked list 
  head can be NULL 
  First element in the linked list is at position 0
  Node is defined as 
  class Node {
     int data;
     Node next;
  }
*/
    

Node InsertNth(Node head, int data, int position) {
   // This is a "method-only" submission. 
    // You only need to complete this method. 
    Node newNode  = new Node();
    newNode.data = data;
    newNode.next = null;
    
    
    if(head == null){
        head = newNode;
    }else{
        if(position == 0){
           newNode.next = head;
           head = newNode; 
           return head;
        }
        int counter = 0;
        Node curr = head;
        Node prev = null;
        while(curr!=null){
           prev = curr;
           curr = curr.next;
           counter++;
           if(counter == position) break;
        }
        prev.next = newNode;
        newNode.next = curr;
    }
  
    return head;
}

File: 30 Days of Code/Day 15/sol.c

Node* newNode(int data);

Node* insert(Node *head,int data)
{
    //Complete this function
    Node* curr = head;
    Node* new = newNode(data);
    if(curr!= NULL){
        curr = head;
        while(curr->next != NULL){
            curr = curr->next;
        }
        curr->next = new;
    }else{ 
        return new;
    }
   return head;
}

/*
 Node* insert(Node *head,int data) {
          Node* newhead = newNode(data);
          if(head != NULL) {
              Node *current = head;
              while(current->next != NULL) {
                  current = current->next;
              }
              current->next = newhead;
              return head;
          } else {
              return newhead;
          }
      }
      */
Node* newNode(int data){
    Node* new = malloc(sizeof(Node));
    new->data = data;
    new->next = NULL;
    return new;
}
